### PR TITLE
fix: Define history only in a browser context

### PR DESCRIPTION
**Pull request by soyuka**
See also #281

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no 
| BC breaks?    | no
| Deprecations? | no
| Tickets       | fixes #... 
| License       | MIT
| Doc PR        | api-platform/docs#...

### CODE REVIEW

**Comment by pawelsmolenski** [comment]
@soyuka Please, try to readjust your code, so checks of this MR will pass... As a motivation I may say that this fix will also fix a problem when history strategy other than "hash" is used. Now, including (indirectly) an `AdminResourcesGuesser` component causes that a hash is always added to the URL on application start in a browser. As a result, this behavior messes the browser history what may and causes a waterfall of other problems – especially on the login page.

In other words: removing obligatory initialization of Hash History by not calling createHashHistory() when another history service is provided will stop HydraAdmin from adding hash to the end of URL on startup.
